### DISCUSSION TITLE
handlebars: compile mode and inheritance

### DISCUSSION THREAD

**Discussion by Bessonov** [webdiscus/html-bundler-webpack-plugin#124] [Q&A] [answered]
Hello everyone,  

I’m a bit stuck and not sure if the issue lies with me, Handlebars, or the bundler. Essentially, I’m trying to replicate [this example](https://handlebarsjs.com/examples/partials/inline-blocks.html):
https://github.com/Bessonov/handlebars-inheritance.

However, when I run:  
```sh  
npm install  
npm start  
```  

I get the following output:  
```
start  
test outside  
end  
```  

Instead of something like:  
```html  
start  
test outside  
<div class="nav">  
  My Nav test navtest layout  
</div>  
<div class="content">  
  My Content  
</div>  
end  
```  

In the (server-side) project, I’m using Webpack’s `entry`, but even when I use the [plugin's entry](https://github.com/Bessonov/handlebars-inheritance/blob/main/webpack.config.mjs#L5), it doesn’t work either.

If I debug the project, I notice [here](https://github.com/webdiscus/html-bundler-webpack-plugin/blob/master/src/Loader/Preprocessors/Handlebars/index.js#L216) that the Handlebars instance contains partials and additional helpers. However, it seems they aren't being compiled into the output.

What am I doing wrong?

BTW, no success with [handlebars-layouts](https://github.com/webdiscus/html-bundler-webpack-plugin/blob/master/test/cases/_preprocessor/handlebars-helper-layouts/webpack.config.js#L41) too.

**Comment by webdiscus** [comment]
Hello @Bessonov,

the Handlebars has two methods:

- compile the template into template function 
- render the template into static HTML

The plugin automatically uses `compile` method for templates imported in JS and uses `render` method for template defined as entrypoints in `entry` plugin option.

The `render` method works 100% for all templates, but `compile` method has limitation and doesn't support all features, e.g. partials ist not supported.

**Reply by webdiscus** [reply]
@Bessonov 

how works plugin, see please tests in my PR

**Reply by webdiscus** [reply]
I'll see what I can do to support partials in compile mode.

**Reply by webdiscus** [reply, marked as answer]
@Bessonov 
I have experimented with precompilation and have a positive result. So very simple partials can be used in compilation mode.

This discussion is transferred to the [[FEATURE REQUEST] handlebars: precompile templates with partials](#126) issue.
